developer/src/kmc-model/src/build-trie.ts

```diff
@@ -291,196 +213,25 @@ namespace Trie {
 
-  function addItemToLeaf(leaf: Leaf, item: Entry) {
-    leaf.entries.push(item);
-  }
-
-  /**
-   * Mutates the given Leaf to turn it into an InternalNode.
-   *
-   * NOTE: the node passed in will be DESTRUCTIVELY CHANGED into a different
-   * type when passed into this function!
-   *
-   * @param depth depth of the trie at this level.
-   */
-  function convertLeafToInternalNode(leaf: Leaf, depth: number): void {
-    let entries = leaf.entries;
-
-    // Alias the current node, as the desired type.
-    let internal = (<unknown> leaf) as InternalNode;
-    internal.type = 'internal';
-
-    delete leaf.entries;
-    internal.values = [];
-    internal.children = {};
-
-    // Convert the old values array into the format for interior nodes.
-    for (let item of entries) {
-      let char: string;
-      if (depth < item.key.length) {
-        char = item.key[depth];
-      } else {
-        char = INTERNAL_VALUE;
-      }
-
-      if (!internal.children[char]) {
-        internal.children[char] = createRootNode();
-        internal.values.push(char);
-      }
-      addUnsorted(internal.children[char], item, depth + 1);
-    }
-
-    internal.unsorted = true;
-  }
-
-  /**
-   * Recursively sort the trie, in descending order of weight.
-   * @param node any node in the trie
-   */
-  function sortTrie(node: Node) {
-    if (node.type === 'leaf') {
-      if (!node.unsorted) {
-        return;
-      }
-
-      node.entries.sort(function (a, b) { return b.weight - a.weight; });
-    } else {
-      // We MUST recurse and sort children before returning.
-      for (let char of node.values) {
-        sortTrie(node.children[char]);
-      }
-
-      if (!node.unsorted) {
-        return;
-      }
-
-      node.values.sort((a, b) => {
-        return node.children[b].weight - node.children[a].weight;
-      });
-    }
-
-    delete node.unsorted;
-  }
-
-  /**
-   * O(n) recursive traversal to sum the total weight of all leaves in the
-   * trie, starting at the provided node.
-   *
-   * @param node The node to start summing weights.
-   */
-  function sumWeights(node: Node): number {
-    let val: number;
-    if (node.type === 'leaf') {
-      val = node.entries
-        .map(entry => entry.weight)
-        //.map(entry => isNaN(entry.weight) ? 1 : entry.weight)
-        .reduce((acc, count) => acc + count, 0);
-    } else {
-      val = Object.keys(node.children)
-        .map((key) => sumWeights(node.children[key]))
-        .reduce((acc, count) => acc + count, 0);
-    }
-
-    if(isNaN(val)) {
-      throw new Error("Unexpected NaN has appeared!");
-    }
-    return val;
+/**
+ * Populates the trie with the contents of an entire wordlist.
+ * @param words a list of word and count pairs.
+ */
+function buildFromWordList(trieCollator: TrieBuilder, words: WordList, toKey: SearchTermToKey): TrieBuilder {
+  for (let [wordform, weight] of Object.entries(words)) {
+    trieCollator.addEntry(wordform, weight);
   }
+  trieCollator.sort();
```

Comment:
This still calls `sortTrie`, just via the object's member function.  (See line 162 of trie-builder.ts.)